### ISSUE TITLE
`taskid` should be optional in problem documents

### ISSUE THREAD

**Issue by tgeoghegan** [ietf-wg-ppm/draft-ietf-ppm-dap#360]
Section 3.2 discusses how to construct HTTP problem documents. In particular, [this paragraph](https://www.ietf.org/archive/id/draft-ietf-ppm-dap-02.html#section-3.2-4) states:

>The problem document MUST also include a "taskid" member which contains the associated DAP task ID (this value is always known, see [Section 4.2](https://www.ietf.org/archive/id/draft-ietf-ppm-dap-02.html#task-configuration)), encoded in Base 64 using the URL and filename safe alphabet with no padding defined in sections 5 and 3.2 of [[RFC4648](https://www.ietf.org/archive/id/draft-ietf-ppm-dap-02.html#RFC4648)].

It's not true that the task ID is always known. [4.3.1](https://www.ietf.org/archive/id/draft-ietf-ppm-dap-02.html#section-4.3.1-3) allows implementations to respond to requests to `/hpke_config` that don't have a `task_id` query parameter with an error, and in that scenario I don't think there is any sensible task ID value for the server to use. We should change this MUST to allow some problem documents to omit `taskid`.

**Comment by tgeoghegan** [comment]
We might also consider making the key `task_id` for consistency with the query parameter on `/hpke_config`.

**Comment by cjpatton** [comment]
I think this is text-ready.

**Comment by tgeoghegan** [comment]
Thinking about this further, I think this is actually a dupe of #278. Currently, we require that problem documents contain the `taskid` and and `instance` value that "MUST be the endpoint to which the request was targeted". The examples of `instance` values in RFC 7807 look like this:

```
{
             "type": "https://example.com/probs/out-of-credit",
             "title": "You do not have enough credit.",
             "detail": "Your current balance is 30, but that costs 50.",
             "instance": "/account/12345/msgs/abc",
             "balance": 30,
             "accounts": ["/account/12345",
                          "/account/67890"]
}
```
Here, the value of `instance` is an HTTP path pointing to the resource that caused the error. Our requirement that `instance` be the "endpoint" seems unhelpful. If I hit some API relative to `https://aggregator.example/dap-api`, and it fails, and I get back a problem document with `"instance": "https://aggregator.example/dap-api"`, then that's not very helpful, because I already knew what URI I accessed when I made the request.

Once we do #278, then I think we won't need a dedicated `taskid` (or `task_id`) field in problem documents anymore, because if the task ID is relevant to the request, then it'll appear in the relative path that we can put in `instance`.

So my plan here is to address #278, and in doing so, update the problem document text. If I don't get around to that for draft-03 for some reason, then the smaller change described here will suffice.
